### ISSUE TITLE
Add support for OAuth access tokens

### ISSUE THREAD

**Issue by GlennV** [mollie/mollie-api-python#99]
By default, OAuth access tokens are not accepted in this package, while other packages do have support (e.g. mollie-api-php). 

I took a look there and found that there's little difference between access_token and API Key calls. The only difference is an additional User Agent header and a hardcoded validation on the API key starting with 'test_' or 'live_' in `validate_api_key()`.

I found a simple workaround to make this package work by avoiding the methods `validate_api_key()` and `set_api_key()`:
 ```python
client = Client()
client.api_key = access_token
client.USER_AGENT += " OAuth/2.0"
```

Maybe it's better to implement this in the package itself (with the necessary validation checks of course).

**Comment by whyscream** [comment]
I think you mean Organisational access token (whose access is defined using Oauth-like properties)? It looks like `access_5qgrcBt3KN9CvTfws7SOMEdxMOREhRANDOMgc`, right?

**Comment by GlennV** [comment]
Not necessarily the Organisational access token, we use Mollie Connect and we need to use our customer's OAuth access_token to query the API. It starts with `access_` indeed.

**Comment by whyscream** [comment]
This package currently does not support any specific endpoints that work only with oAuth such as [Connect API](https://docs.mollie.com/reference/oauth2/authorize), [Organisation API](https://docs.mollie.com/reference/v2/organizations-api/current-organization) and [Profile API](https://docs.mollie.com/reference/v2/profiles-api/create-profile).

Can you specify a bit more in detail which endpoints you intend to use, and for what purpose?

**Comment by GlennV** [comment]
Sure, we just need the Payments API so our customers (we are a CRM for non-profits) can use our Mollie App to let their members pay online and we can charge an Application Fee.

The Connect API is just used for getting & refreshing the access tokens, which we already handle with the brilliant https://github.com/requests/requests-oauthlib package.

**Comment by willemstuursma** [comment]
@whyscream most endpoints that allow API keys, can also be used with organization or access tokens.

**Comment by whyscream** [comment]
@GlennV we just released v2.1.1, this version has support for configuring access token using`client.set_access_token('access_12345')`.
